### PR TITLE
Free knitro in tests and split lic man tests

### LINKED ISSUE #125: Can't run test until the end (Network license)
I a having problem running the KNITRO tests in julia. The problem is, my KNITRO is not releasing automatically the network license.
Probably calling the function that frees the licenses after each solve is enough for these network licenses.

The output of:
```julia
] test KNITRO
```
is:
```julia
=======================================
          Commercial License
         Artelys Knitro 12.0.0
=======================================

Knitro presolve eliminated 0 variables and 0 constraints.

hessopt:                 6
The problem is identified as bound constrained only.
Knitro changing algorithm from AUTO to 1.
Knitro changing bar_initpt from AUTO to 3.
Knitro changing bar_murule from AUTO to 4.
Knitro changing bar_penaltycons from AUTO to 0.
Knitro changing bar_penaltyrule from AUTO to 2.
Knitro changing bar_switchrule from AUTO to 0.
Knitro changing linesearch from AUTO to 2.
Knitro changing linsolver from AUTO to 2.

Problem Characteristics                                 (   Presolved)
-----------------------
Objective goal:  Minimize
Objective type:  general
Number of variables:                                  2 (           2)
    bounded below only:                               0 (           0)
    bounded above only:                               0 (           0)
    bounded below and above:                          2 (           2)
    fixed:                                            0 (           0)
    free:                                             0 (           0)
Number of constraints:                                0 (           0)
    linear equalities:                                0 (           0)
    quadratic equalities:                             0 (           0)
    gen. nonlinear equalities:                        0 (           0)
    linear one-sided inequalities:                    0 (           0)
    quadratic one-sided inequalities:                 0 (           0)
    gen. nonlinear one-sided inequalities:            0 (           0)
    linear two-sided inequalities:                    0 (           0)
    quadratic two-sided inequalities:                 0 (           0)
    gen. nonlinear two-sided inequalities:            0 (           0)
Number of nonzeros in Jacobian:                       0 (           0)
Number of nonzeros in Hessian:                        0 (           0)

  Iter      Objective      FeasError   OptError    ||Step||    CGits
--------  --------------  ----------  ----------  ----------  -------
       0    1.434839e+00   0.000e+00
       7    2.310122e-15   0.000e+00   7.975e-08   4.158e-06        0

EXIT: Locally optimal solution found.

Final Statistics
----------------
Final objective value               =   2.31012172961205e-15
Final feasibility error (abs / rel) =   0.00e+00 / 0.00e+00
Final optimality error  (abs / rel) =   7.97e-08 / 7.97e-08
# of iterations                     =          7
# of CG iterations                  =          0
# of function evaluations           =          8
# of gradient evaluations           =          8
Total program time (secs)           =       0.042 (     0.016 CPU time)
Time spent in evaluations (secs)    =       0.000

===============================================================================

### License found, but all copies are checked out
###   Could not find a valid license on the server.
### No license found
###   Could not find a valid license on the server.
### Could not find a valid license.
    Your machine ID is 00-b3-70-9f-2b.
    Please contact licensing@artelys.com or your local distributor to obtain a license.
    If you already have a license, please execute `get_machine_ID.exe -v` and send the output to support.
    Go to http://www.artelys.com/knitro for a Knitro student version license or trial.
Test old API: Error During Test at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\test\runtests.jl:5
  Got exception outside of a @test
  LoadError: LoadError: KNITRO: Error creating solver
  Stacktrace:
   [1] error(::String) at .\error.jl:33
   [2] newcontext at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\src\ktr_functions.jl:32 [inlined]
   [3] Type at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\src\ktr_model.jl:39 [inlined]
   [4] createProblem() at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\src\ktr_model.jl:48
   [5] top-level scope at none:0
   [6] include at .\boot.jl:326 [inlined]
   [7] include_relative(::Module, ::String) at .\loading.jl:1038
   [8] include(::Module, ::String) at .\sysimg.jl:29
   [9] include(::String) at .\client.jl:403
   [10] top-level scope at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\test\oldexamples.jl:4 [inlined]
   [11] top-level scope at .\none:0
   [12] include at .\boot.jl:326 [inlined]
   [13] include_relative(::Module, ::String) at .\loading.jl:1038
   [14] include(::Module, ::String) at .\sysimg.jl:29
   [15] include(::String) at .\client.jl:403
   [16] top-level scope at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\test\runtests.jl:6
   [17] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
   [18] top-level scope at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\test\runtests.jl:6
   [19] include at .\boot.jl:326 [inlined]
   [20] include_relative(::Module, ::String) at .\loading.jl:1038
   [21] include(::Module, ::String) at .\sysimg.jl:29
   [22] include(::String) at .\client.jl:403
   [23] top-level scope at none:0
   [24] eval(::Module, ::Any) at .\boot.jl:328
   [25] exec_options(::Base.JLOptions) at .\client.jl:243
   [26] _start() at .\client.jl:436
  in expression starting at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\examples\oldinterface\hs015_tuner.jl:74
  in expression starting at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\test\oldexamples.jl:1
Test Summary:            | Pass  Error  Total
Test old API             |    2      1      3
  Test optimal solutions |    2             2
ERROR: LoadError: Some tests did not pass: 2 passed, 0 failed, 1 errored, 0 broken.
in expression starting at C:\Users\joaquimgarcia\.julia\packages\KNITRO\jIIje\test\runtests.jl:5
ERROR: Package KNITRO errored during testing
```

### PR DESCRIPTION
**Pull request by joaquimg**
closes #125 

I added "free" as an attribute to be easily accessible via JuMP.

Also added a try catch in the lic manager tests which is not supported by all users.

### CODE REVIEW

File: examples/nlp1.jl [hunk 1]
```diff
@@ -192,7 +192,7 @@ function example_nlp1()
     println("  KKT optimality violation = ", KNITRO.KN_get_abs_opt_error(kc))
 
     # Delete the Knitro solver instance.
-    #= KNITRO.KN_free(kc) =#
+    KNITRO.KN_free(kc)
```
Comment (frapac) [inline examples/nlp1.jl:195]: Good catch. This should not have been commented out

File: test/runtests.jl [hunk 2]
```diff
@@ -29,4 +29,17 @@ if KNITRO.KNITRO_VERSION >= v"11.0"
     @testset "Test JuMP" begin
         include("jump_soc.jl")
     end
+    try
+        @testset "License tests" begin
+            @testset "Test C API" begin
```
Comment (frapac) [inline test/runtests.jl:34]: I would maybe remove the additional `@testset` there

File: test/MOI_wrapper.jl [hunk 3]
```diff
@@ -21,29 +21,27 @@ const config_noduals = MOIT.TestConfig(atol=1e-5, rtol=1e-8,
...
-
-# Build bridge optimizer.
-const BRIDGED = MOIB.full_bridge_optimizer(OPTIMIZER, Float64)
-
 @testset "MOI utils" begin
     @testset "SolverName" begin
         optimizer = KNITRO.Optimizer()
         @test MOI.get(optimizer, MOI.SolverName()) == "Knitro"
+        MOI.empty!(optimizer)
+        KNITRO.KN_free(optimizer.inner)
```
Comment (frapac) [inline test/MOI_wrapper.jl:29]: I am not sure to understand why we need to call `KN_free` here? Haven't we freed the model already in `MOI.empty!`?

File: test/MOI_wrapper.jl [hunk 4]
```diff
@@ -21,29 +21,27 @@ const config_noduals = MOIT.TestConfig(atol=1e-5, rtol=1e-8,
...
     end
     @testset "supports_default_copy_to" begin
         optimizer = KNITRO.Optimizer()
         @test MOIU.supports_default_copy_to(optimizer, false)
         # Use `@test !...` if names are not supported
         @test MOIU.supports_default_copy_to(optimizer, true)
+        KNITRO.KN_free(optimizer.inner)
```
Comment (frapac) [inline test/MOI_wrapper.jl:36]: I would replace `KN_free` by `MOI.empty!` there

**Review by frapac — CHANGES_REQUESTED** [review]
Very good catch! 
I think this new option `free` is a nice add-up to Knitro. 
About the PR, I think it is ready to merge as soon we have decided whether to use `KN_free` or `MOI.empty!` in the MOI's tests.

**Comment by joaquimg** [comment]
I will switch to KNITRO.free when feasible.
MOI.empty! Is no enough because it frees a model and creates a new one so that you can start adding data.

**Review by frapac — APPROVED** [review]
It makes sense to use only the function `KNITRO.free`, then. The changes look good to me.

**Pull request merged**